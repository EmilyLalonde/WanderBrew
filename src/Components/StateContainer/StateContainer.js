import React from 'react'
import { connect } from 'react-redux';
import './StateContainer.css'
import Card from '../Card/Card'

const StateContainer = ({stateResults}) => {
  let searchResults = stateResults.map(stateObj => {
      return <Card 
      name={stateObj.name}
      street={stateObj.street} 
      city={stateObj.city}
      state={stateObj.state}
      phone={stateObj.phone}
      key={stateObj.id}
      />
  })
  return (
    <div>
      {searchResults}
    </div>
  )
}

const mapStateToProps = state => ({
  stateResults:state.stateResults
})

export default connect(mapStateToProps, null)(StateContainer)