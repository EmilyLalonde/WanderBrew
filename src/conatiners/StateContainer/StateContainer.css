.state-search-results {
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  margin-top: 3%;
} 