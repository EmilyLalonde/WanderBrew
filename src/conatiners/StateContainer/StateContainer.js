import React from 'react'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import './StateContainer.css'
import Card from '../../Components/Card/Card'

const StateContainer = ({ stateResults }) => {
  let searchResults = stateResults.map(stateObj => {
      return (
      <Link to={`/state-search/${stateObj.id}`} key={stateObj.id}>
      <Card 
      name={stateObj.name}
      street={stateObj.street} 
      city={stateObj.city}
      state={stateObj.state}
      phone={stateObj.phone}
      key={stateObj.id}
      id={stateObj.id}
      />
      </Link>
      )
  })
  return (
    <div>
      {searchResults}
    </div>
  )
}

const mapStateToProps = state => ({
  stateResults:state.stateResults
})

export default connect(mapStateToProps, null)(StateContainer)