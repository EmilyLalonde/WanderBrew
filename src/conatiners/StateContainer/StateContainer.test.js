import React from 'react'
import { shallow } from 'enzyme'
import {StateContainer} from './StateContainer'

describe('StateContainer', () => {
  let wrapper;
  let mockBreweries;

  beforeEach(() => {
    mockBreweries = [{id: 1516, name: 'Ratio Beerworks', street: '2920 Larimer St', city: 'Denver', state:'Colorado', phone:'3012669264'}]
    wrapper= shallow(<StateContainer nameResults={mockBreweries}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})