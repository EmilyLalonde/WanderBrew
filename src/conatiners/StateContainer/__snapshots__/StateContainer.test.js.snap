// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateContainer should match snapshot 1`] = `undefined`;
